src/cache_cf.cc

```diff
@@ -3072,7 +3072,8 @@ free_time_msec(time_msec_t * var)
 static void
 dump_time_nanoseconds(StoreEntry *entry, const char *name, const std::chrono::nanoseconds &var)
 {
-    storeAppendPrintf(entry, "%s %" PRId64 " nanoseconds\n", name, var.count());
+    // This assumes that nothing in C++ is larger than long long. TODO: Avoid printf().
+    storeAppendPrintf(entry, "%s %lld nanoseconds\n", name, static_cast<long long>(var.count()));
```

Comment:
Sorry. While syncing the PR description with the PR code, I realized that we may be able to do even better:

```suggestion
    // std::chrono::nanoseconds::rep is unknown a priori so we cast to (and print) the largest supported integer
    storeAppendPrintf(entry, "%s %jd nanoseconds\n", name, static_cast<intmax_t>(var.count()));
```

The above should work in all cases as long as the underlying printf() supports `%jd` which was added to [C++11](https://en.cppreference.com/w/cpp/io/c/fprintf). Again, I do not know how portable that is in Squid-relevant environments.

If you accept this suggestion, please adjust the PR description accordingly. If you reject it, please check that you are OK with the current PR description.

Reply:
There [is](https://en.cppreference.com/w/c/types/integer) also `PRIdMAX`. I do not know whether it is any more portable than raw `%jd`.